daemons/ipa-kdb/ipa_kdb_principals.c

```diff
@@ -327,7 +327,7 @@ static void ipadb_validate_password(struct ipadb_context *ipactx,
         *ua |= IPADB_USER_AUTH_PASSWORD;
 }
 
-static enum ipadb_user_auth ipadb_get_user_auth(struct ipadb_context *ipactx,
+enum ipadb_user_auth ipadb_get_user_auth(struct ipadb_context *ipactx,
```

Comment:
you won't have to change this if we always unconditionally add this data to struct ipadb_e_data and use ipadb_get_principal() as mentioned above